lib/types/object/index.js

```diff
@@ -118,16 +118,32 @@ internals.Object = class extends Any {
                 }
             }
 
-            if (target[rename.from] === undefined) {
+
+            const isRegExRename = rename.from instanceof RegExp;
+
+            if (target[rename.from] === undefined && !isRegExRename) {
                 delete target[rename.to];
             }
+            else if (isRegExRename) {
+                const objectKeys = Object.keys(target);
+
+                for (let j = 0; j < objectKeys.length; ++j) {
+                    if (rename.from.test(objectKeys[j])) {
+                        target[rename.to] = target[objectKeys[j]];
+
+                        if (!rename.options.override) {
```

Comment:
From what I see in the docs this is not the correct usage of `override`. 

> `override` - if `true`, allows renaming a key over an existing key. Defaults to `false`.

Instead, this particular usage should be `alias`.

> `alias` - if `true`, does not delete the old key name, keeping both the new and old keys in place. Defaults to `false`.

So this particular line should be `if (!rename.options.alias) {`

and then we also need to implement `override` for the regular expressions as well.

You can see how we implement that [here](https://github.com/hapijs/joi/blob/069bb6c04063cba202544dacf9da3bbe21fa3074/lib/types/object/index.js#L111-L119).